ReactAndroid/src/main/java/com/facebook/react/modules/settings/SettingsModule.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.modules.settings;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.ReadableMapKeySetIterator;
+import com.facebook.react.bridge.ReadableType;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.module.annotations.ReactModule;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@ReactModule(name = SettingsModule.NAME)
+public class SettingsModule extends ReactContextBaseJavaModule  implements SharedPreferences.OnSharedPreferenceChangeListener{
+
+  @SuppressWarnings("WeakerAccess")
+  static final String NAME = "SettingsManager";
+  static final String DEFAULT_FILE_NAME = "ReactNative";
+
+  static private String sFilename = "";
+  private Boolean ignoringUpdates = false;
+  private SharedPreferences mPreferences;
+
+  public SettingsModule(ReactApplicationContext context) {
+    super(context);
+  }
+
+  @Override
+  public String getName() {
+    return SettingsModule.NAME;
+  }
+
+  private SharedPreferences getPreferences() {
+    if (mPreferences == null) {
+      if (SettingsModule.sFilename.length() > 0) {
+        mPreferences = getReactApplicationContext().getSharedPreferences(SettingsModule.sFilename, Context.MODE_PRIVATE);
+      } else if (getCurrentActivity() != null) {
+        mPreferences = getCurrentActivity().getPreferences(Context.MODE_PRIVATE);
```

Comment:
Yes, and differing calls here is entirely intentional. `getSharedPreferences()`  lets the caller specify a specific file name in which to store the shared preferences, whereas `getPreferences()` stores the preferences values in a file with a default name.

In the section above we allow the programmer the option of storing the preferences in a file with a specific name (say, to match an existing shared preferences file already in use by the application), or, to simply store the preferences in a file with a default name.